command.go

```diff
@@ -0,0 +1,234 @@
+package ff
+
+import (
+	"context"
+	"fmt"
+	"strings"
+)
+
+// Command is a declarative structure that combines a main function with a flag
+// set and zero or more subcommands. It's intended to model CLI applications
+// which can be be represented as a tree of such commands.
+type Command struct {
+	// Name of the command, which is used when producing the help output for the
+	// command, as well as for subcommand matching.
+	//
+	// Required.
+	Name string
+
+	// Usage is a single line which should describe the syntax of the command,
+	// including flags and arguments. It's typically printed at the top of the
+	// help output for the command. For example,
+	//
+	//    USAGE
+	//      cmd [FLAGS] subcmd [FLAGS] <ARG> [<ARG>...]
+	//
+	// Here, the usage string begins with "cmd [FLAGS] ...".
+	//
+	// Recommended. If not provided, the help output for the command should not
+	// include a usage section.
+	Usage string
+
+	// ShortHelp is a single line which should very briefly describe the purpose
+	// of the command in prose. It's typically printed next to the command name
+	// when it appears as a subcommand in help output. For example,
+	//
+	//    SUBCOMMANDS
+	//      commandname   this is the short help string
+	//
+	// Recommended.
+	ShortHelp string
+
+	// LongHelp is a multi-line string, usually one or more paragraphs of prose,
+	// which explain the command in detail. It's typically included in the help
+	// output for the command, separate from other sections.
+	//
+	// Long help should be formatted for user readability. For example, if help
+	// output is written to a terminal, long help should hard-wrap lines at an
+	// appropriate column width for that terminal.
+	//
+	// Optional.
+	LongHelp string
+
+	// Flags is the set of flags associated with, and parsed by, this command.
+	//
+	// When building a command tree, it's often useful to allow flags defined by
+	// parent commands to be specified by any subcommand. The core flag set
+	// supports this behavior via SetParent, see the documentation of that
+	// method for details.
+	//
+	// Optional. If not provided, an empty flag set will be constructed and used
+	// so that the -h, --help flag works as expected.
+	Flags Flags
```

Comment:
Command.Flags needs to be a true Flags, meaning you can't pass a flag.FlagSet directly, you have to use the adapter constructor. I think this is reasonable: ff.Parse is meant to be symmetrical with flag.Parse, but ff.Command doesn't have any direct analogue in that sense.